src/Silverback.Integration.Kafka/Messaging/Broker/KafkaConsumer.cs

```diff
@@ -127,7 +127,7 @@ internal void OnPartitionsRevoked()
 
             AsyncHelper.RunSynchronously(WaitUntilChannelsManagerStopsAsync);
 
-            if (!Endpoint.Configuration.IsAutoCommitEnabled)
+            if (!Endpoint.Configuration.IsAutoCommitEnabled && Endpoint.Configuration.IsGroupIdSet)
```

Comment:
Create a method to encapsulate this logic (eg. `IsCommitNecessary()`) or move the check into the `CommitOffsets` method directly? The condition is duplicated below.